### PR TITLE
Adding test case for SharedBackupEdit

### PR DESCRIPTION
**Pull request by skonda-px**
**What this PR does / why we need it**:
  Adding Test case for Shared Backup Edit
  https://portworx.testrail.net/index.php?/cases/view/82950

**Which issue(s) this PR fixes** (optional)
Closes #PA-506

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/backup/backup_test.go [hunk 1]
```diff
@@ -899,6 +899,241 @@ var _ = Describe("{ShareBackupWithUsersAndGroups}", func() {
...
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+	It("Share large number of backups", func() {
```
Comment (kshithijiyer-px) [inline tests/backup/backup_test.go:938]: Correct the It string here according to you testcase please.

File: tests/backup/backup_test.go [hunk 2]
```diff
@@ -899,6 +899,241 @@ var _ = Describe("{ShareBackupWithUsersAndGroups}", func() {
...
+					wg.Done()
+				}(userName, firstName, lastName, email)
+			}
+			wg.Wait()
+		})
+		Step("Adding Credentials and Registering Backup Location", func() {
+			log.InfoD("Creating bucket, cloud credentials and backup location")
+			for _, provider := range providers {
+				customBucketName = fmt.Sprintf("%s-%v", getBucketName()[0], time.Now().Unix())
+				CreateBucket(provider, customBucketName)
```
Comment (kshithijiyer-px) [inline tests/backup/backup_test.go:966]: Why do you want a special bucket here? Can we use the common bucket?

File: tests/backup/backup_test.go [hunk 3]
```diff
@@ -899,6 +899,241 @@ var _ = Describe("{ShareBackupWithUsersAndGroups}", func() {
...
+		log.Infof("Deleting registered clusters for non-admin context")
+		for _, ctxNonAdmin := range userContexts {
+			DeleteCluster(SourceClusterName, orgID, ctxNonAdmin)
+			DeleteCluster(destinationClusterName, orgID, ctxNonAdmin)
+		}
+
+		log.Infof("Cleaning up backup location - %s", customBackupLocationName)
+		DeleteBackupLocation(customBackupLocationName, backupLocationUID, orgID)
+
+		log.Infof("Cleaning up buckets")
```
Comment (kshithijiyer-px) [inline tests/backup/backup_test.go:1124]: Not required if you use the common bucket

**Review by kshithijiyer-px — CHANGES_REQUESTED** [review]

File: tests/backup/backup_test.go (hunk 2 above)
Comment (skonda-px) [reply tests/backup/backup_test.go]: Changed the bucked to common bucket

**Review by kshithijiyer-px — APPROVED** [review]

**Comment by kshithijiyer-px** [comment]
@skonda-px Please share the atos dashboard link

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Comment by skonda-px** [comment]
https://aetos.pwx.purestorage.com/resultSet/testSetID/76584

**Comment by skonda-px** [comment]
after addressing review comments : http://aetos.pwx.purestorage.com/resultSet/testSetID/76668

**Pull request merged**